src/Quartz/CronExpression.cs

```diff
@@ -209,45 +209,68 @@ namespace Quartz
     [Serializable]
     public class CronExpression : IDeserializationCallback, ISerializable
     {
-        private static readonly Dictionary<string, int> monthMap = new Dictionary<string, int>(20);
-        private static readonly Dictionary<string, int> dayMap = new Dictionary<string, int>(60);
+        private static readonly IReadOnlyDictionary<string, int> monthMap = new Dictionary<string,int>(12)
+        {
+            {"JAN", 0},
+            {"FEB", 1},
+            {"MAR", 2},
+            {"APR", 3},
+            {"MAY", 4},
+            {"JUN", 5},
+            {"JUL", 6},
+            {"AUG", 7},
+            {"SEP", 8},
+            {"OCT", 9},
+            {"NOV", 10},
+            {"DEC", 11}
+        };
+        private static readonly IReadOnlyDictionary<string, int> dayMap = new Dictionary<string, int>(7)
+        {
+            {"SUN", 1},
+            {"MON", 2},
+            {"TUE", 3},
+            {"WED", 4},
+            {"THU", 5},
+            {"FRI", 6},
+            {"SAT", 7}
+        };
 
         private TimeZoneInfo? timeZone;
 
         /// <summary>
         /// Seconds.
         /// </summary>
-        [NonSerialized] protected SortedSet<int> seconds = null!;
+        [NonSerialized] protected SortedSet<int> seconds = new();
```

Comment:
if everything is new'ed, there's allocation cost from unused members, are they always needed?

Reply:
Constructor calls through to BuildExpression(...)
So in all cases they are all used because they all initialise to new() in the constructor
But the original justification for the change to new up the members was this:

![image](https://user-images.githubusercontent.com/127927/230362391-36251f37-29ff-4b94-a392-5865b07a1406.png)


(and probably the tendency for me to avoid nulls where I can, obviously not factoring in performance concerns though)

I'll revert back to the above along with setting to members to `null!` instead of `new()`